### DISCUSSION TITLE
Docs clarification for I/O

### DISCUSSION THREAD

**Discussion by scottyhq** [corteva/rioxarray#538] [Ideas]
I'm wondering if the getting started page (https://corteva.github.io/rioxarray/stable/getting_started/getting_started.html#rioxarray )  can be updated now that xarray has `DeprecationWarning: open_rasterio is Deprecated in favor of rioxarray`

1. Is it true that `rioxarray.open_rasterio('geo.tif')` == `xarray.open_dataarray('geo.tif', engine='rasterio')`? 

2. Why use [rioxarray.open_rasterio()](https://corteva.github.io/rioxarray/stable/rioxarray.html#rioxarray.open_rasterio) instead of xarray.open_rasterio? --> What does the 'rasterio' engine do when opening files?

**Comment by snowman2** [comment]
> Is it true that rioxarray.open_rasterio('geo.tif') == xarray.open_dataarray('geo.tif', engine='rasterio')?

For TIF files, that is the case. However, for netCDF files or multidimensional files, rioxarray.open_rasterio('geo.nc') == xarray.open_dataset('geo.nc', engine='rasterio'). 

See also #495 

> Why use [rioxarray.open_rasterio()](https://corteva.github.io/rioxarray/stable/rioxarray.html#rioxarray.open_rasterio) instead of xarray.open_rasterio? --> What does the 'rasterio' engine do when opening files?

:+1: - sounds good to me.

**Comment by scottyhq** [comment]
passing `backend_kwargs` doesn't quite work as expected unless i'm missing something. cc @dcherian 
```python
da = xr.open_dataarray('https://github.com/corteva/rioxarray/raw/master/test/test_data/input/cog.tif',
                       engine='rasterio', 
                       backend_kwargs=dict(masked=True, overview_level=4),
                      )
```

```pytb
File ~/miniconda3/envs/xarray-tutorial/lib/python3.10/site-packages/xarray/backends/api.py:495, in open_dataset(filename_or_obj, engine, chunks, cache, decode_cf, mask_and_scale, decode_times, decode_timedelta, use_cftime, concat_characters, decode_coords, drop_variables, backend_kwargs, *args, **kwargs)
    483 decoders = _resolve_decoders_kwargs(
    484     decode_cf,
    485     open_backend_dataset_parameters=backend.open_dataset_parameters,
   (...)
    491     decode_coords=decode_coords,
    492 )
    494 overwrite_encoded_chunks = kwargs.pop("overwrite_encoded_chunks", None)
--> 495 backend_ds = backend.open_dataset(
    496     filename_or_obj,
    497     drop_variables=drop_variables,
    498     **decoders,
    499     **kwargs,
    500 )
    501 ds = _dataset_from_backend_dataset(
    502     backend_ds,
    503     filename_or_obj,
   (...)
    510     **kwargs,
    511 )
    512 return ds

TypeError: RasterioBackend.open_dataset() got an unexpected keyword argument 'overview_level'
```

**Reply by snowman2** [reply]
Looks like it needs to be `open_kwargs={"overview_level": 4}` [ref](https://github.com/corteva/rioxarray/blob/570150fbb0d359c5567a08a71adb5efc908164f0/rioxarray/xarray_plugin.py#L51). Probably useful to add that to the docs as well.

**Reply by dcherian** [reply]
Alternatively, could change to ``**open_kwargs`` in https://github.com/corteva/rioxarray/blob/570150fbb0d359c5567a08a71adb5efc908164f0/rioxarray/xarray_plugin.py#L36-L52

**Reply by dcherian** [reply]
Also https://docs.xarray.dev/en/stable/internals/how-to-add-new-backend.html#backendentrypoint-subclassing says "chunks" and "cache" are handled by Xarray so they should be unnecessary.

**Reply by snowman2** [reply]
> Also https://docs.xarray.dev/en/stable/internals/how-to-add-new-backend.html#backendentrypoint-subclassing says "chunks" and "cache" are handled by Xarray so they should be unnecessary.

We tried that. Unfortunately, the way xarray handles it, it isn't possible.
